### PR TITLE
add integration tests to ci build

### PR DESCRIPTION
**Pull request by cosmicBboy**
Signed-off-by: Niels Bantilan <niels.bantilan@gmail.com>

### CODE REVIEW

File: tests/integration/test_flyte_remote.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            return fn()
+        except _InactiveRpcError as exc:
+            if not exc.details().startswith(RETRY_ERROR):
+                raise
+            time.sleep(wait_time)
+
+
+# disable this test on python 3.7, see:
+# https://stackoverflow.com/questions/63329657/python-3-7-error-unsupported-pickle-protocol-5
+@pytest.mark.skipif(
```
Comment (cosmicBboy) [inline tests/integration/test_flyte_remote.py:83]: see: https://github.com/unionai-oss/unionml/actions/runs/3116433995/jobs/5054258085#step:9:10928

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -35,14 +41,23 @@ jobs:
...
         pre-commit run --all-files --show-diff-on-failure
-    - name: Unit tests with pytest
+    - name: Install FlyteCTL
+      if: ${{ matrix.test-dir == 'integration' }}
+      uses: unionai-oss/flytectl-setup-action@master
+    - name: Setup Flyte Sandbox
+      if: ${{ matrix.test-dir == 'integration' }}
+      run: |
+        flytectl sandbox start --source .
+        flytectl config init
```
Comment (eapolinario) [inline .github/workflows/build.yml:57]: do we need this now that we have the code to setup the sandbox as a fixture?

**Review by eapolinario — APPROVED** [review]
Just to leave this recorded, session fixtures and pytest-xdist do not play super well: https://github.com/pytest-dev/pytest/issues/252

The sandbox had to be programmable in order for us to rely on that, otherwise, LGTM.

**Comment by cosmicBboy** [comment]
> Just to leave this recorded, session fixtures and pytest-xdist do not play super well: [pytest-dev/pytest#252](https://github.com/pytest-dev/pytest/issues/252)
> 
> The sandbox had to be programmable in order for us to rely on that, otherwise, LGTM.

yep, learned that just now :)

**Pull request merged**